R/compute_centiloid.R

```diff
@@ -0,0 +1,176 @@
+#'   When custom parameters are not provided, accepted values are:
+#'   '18F-Florbetapir', '18F-Florbetaben'
+#'   If custom slope and intercept are provided, any string is accepted.
+#'
+#' @param pipeline SUVR pipeline from nv_neuro dataset
+#'
+#'   A character string is expected.
+#'   When custom parameters are not provided, accepted values are:
+#'   'AVID FBP SUVR PIPELINE', 'BERKELEY FBP SUVR PIPELINE', 'BERKELEY FBB SUVR PIPELINE'
+#'   If custom slope and intercept are provided, any string is accepted.
+#'
+#' @param ref_region Reference region from suppnv_neuro dataset
+#'
+#'   A character string is expected.
+#'   When custom parameters are not provided, accepted values are:
+#'   'Whole Cerebellum', 'Composite Reference Region'
+#'   If custom slope and intercept are provided, any string is accepted.
+#'
+#' @param suvr SUVR value
+#'
+#'   A numeric value is expected.
+#'
+#' @param custom_slope Optional slope parameter for custom centiloid calculation formula
+#'
+#'   A numeric value is expected when provided.
+#'   When provided with custom_slope, this overrides the standard formula parameters.
+#'   Default is NULL.
+#'
+#' @param custom_intercept Optional intercept parameter for custom centiloid calculation formula
+#'
+#'   A numeric value is expected when provided.
+#'   When provided with custom_intercept, this overrides the standard formula parameters.
+#'   Default is NULL.
+#'
+#' @details
+#' The centiloid scale is a standardized quantitative measure for amyloid PET imaging
+#' that allows comparison between different tracers and analysis methods. This function
+#' converts SUVR values to the centiloid scale based on published conversion
+#' equations for specific tracer, pipeline, and reference region combinations.
+#'
+#' Centiloid is calculated as:
+#'
+#' Centiloid = slope * SUVR + intercept
+#'
+#' where slope and intercept are formula parameters. If custom slope and intercept are provided,
+#' this formula is used regardless of the tracer, pipeline, or reference region values.
+#'
+#' If a matching standard combination is not found and no custom parameters are provided,
+#' a warning is issued and `NA_real_` is returned.
+#'
+#' @return A numeric centiloid value
+#'
+#' @keywords com_bds_findings
+#' @family com_bds_findings
+#'
+#' @export
+#'
+#' @examples
+#' # Using standard parameters
+#' compute_centiloid(
+#'   tracer = "18F-Florbetapir",
+#'   pipeline = "AVID FBP SUVR PIPELINE",
+#'   ref_region = "Whole Cerebellum",
+#'   suvr = 1.25
+#' )
+#'
+#' # Using custom parameters
+#' compute_centiloid(
+#'   tracer = "MyTracer",
+#'   pipeline = "MyPipeline",
+#'   ref_region = "MyRegion",
+#'   suvr = 1.25,
+#'   custom_slope = 193,
+#'   custom_intercept = -187
+#' )
+compute_centiloid <- function(
+    tracer,
+    pipeline,
+    ref_region,
+    suvr,
+    custom_slope = NULL,
+    custom_intercept = NULL) {
+  params_map <- list(
+    # reference: https://doi.org/10.1016/j.jalz.2018.06.1353
```

Comment:
Ok to have references here, but these comments are not exposed to the users unless they peruse the actual function code (unlikely). As such, please add each of the three references to the `@details` section as well, as the roxygen documentation is exposed to the users.